src/current/v24.3/troubleshoot-lock-contention.md

```diff
@@ -0,0 +1,395 @@
+
+<img src="{{ 'images/v24.3/troubleshoot-lock-contention-blocking-transaction-details.png' | relative_url }}" alt="Blocking transaction details" style="border:1px solid #eee;max-width:100%" />
+
+### Additional practice
+
+For Transaction 3, take steps similar to the above steps for Transaction 2, to identify the waiting statement that experienced high contention and the corresponding blocking transaction.
+
+## Remediate lock contention
+
+### Background context
+
+Locking conflicts are a natural artifact when business requirements call for concurrent data changes. Realistically, locking conflicts are unavoidable. The locking conflicts, however, are resolved efficiently with regard to the underlying resource utilization. When blocked transactions are waiting on a lock, they are not consuming CPU, disk, or network resources.
+
+Remediation is required when locking conflicts are too numerous, resulting in a significant increase in response time and/or decrease in throughput. Remediation of locking conflicts is typically about giving up some functionality in exchange for a reduction in locking contention. [Example 2](#example-2) uses two ways of doing this: [historical queries]({% link {{ page.version.version }}/as-of-system-time.md %}) and a ["fail fast" method]({% link {{ page.version.version }}/select-for-update.md %}#wait-policies). Use these remediations if they fit your application design. Other possible ways to reduce lock conflicts are: [column families]({% link {{ page.version.version }}/column-families.md %}) and [secondary indexes]({% link {{ page.version.version }}/schema-design-indexes.md %}).
+
+### Historical queries
+
+In this tutorial's [Example 2](#example-2), Transaction 5 uses a historical query:
+
+~~~ sql
+BEGIN AS OF SYSTEM TIME '-30s';
+SELECT * FROM t2 WHERE k=4; -- historical read
+COMMIT;
+~~~
+
+Consider the following when using [historical queries]({% link {{ page.version.version }}/as-of-system-time.md %}):
+
+- Use historical queries only if the application can use data that is 5 seconds old or older.
+- Historical queries primarily benefit read-only transactions.
+- Historical queries operate below [closed timestamps]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#closed-timestamps) and therefore have perfect concurrency characteristics - they never wait on anything and never block anything.
+- Historical queries have the best possible performance, since they are served by the nearest replica.
+
+### &quot;fail fast&quot; method
+
+In this tutorial's [Example 2](#example-2), Transaction 6 uses a &quot;fail fast&quot; method:
+
+~~~ sql
+BEGIN;
+SELECT * FROM t2 WHERE k=4 FOR UPDATE NOWAIT; -- fail fast write
+UPDATE t2 SET v=4034 WHERE k=4;
+COMMIT;
+~~~
+
+Consider the following when using a ["fail fast" method]({% link {{ page.version.version }}/select-for-update.md %}#wait-policies):
```

Comment:
this is confusingly worded, since i thought this was the 'fail fast' section but this is linking to the SELECT FOR UPDATE doc which doesn't say anything about "fail fast"

suggest revising this to make clear it's an external link to the statement docs, something like

> Consider the following when using  a "fail fast" method such as [`SELECT FOR UPDATE ... NOWAIT`]({% link {{ page.version.version }}/select-for-update.md %}#wait-policies):

Reply:
removed this sentence and updated the section